Mathlib/CategoryTheory/Category/Cat/Adjunction.lean

```diff
@@ -51,4 +49,48 @@ def typeToCatObjectsAdj : typeToCat ⊣ Cat.objects where
           obtain rfl := Discrete.eq_of_hom f
           aesop_cat ) }
 
+
+/-- The connected components functor -/
+def connectedComponents.{v} : Cat.{v, u} ⥤ Type u where
+  obj C := ConnectedComponents C
+  map F :=
+    Quotient.lift (Quotient.mk (Zigzag.setoid _) ∘ F.obj)
+      (fun _ _ ↦ Quot.sound ∘ zigzag_obj_of_zigzag F)
+  map_id _ := funext fun x ↦ (Quotient.exists_rep x).elim (fun _ h ↦ by simp [<- h]; rfl)
+  map_comp _ _ := funext fun x ↦ (Quotient.exists_rep x).elim (fun _ h => by simp [<- h])
+
+
+private def fnToFctr  {X C} (fct : connectedComponents.obj C ⟶ X) : (C ⥤ typeToCat.obj X) where
+  obj :=  Discrete.mk ∘ fct ∘ Quotient.mk (Zigzag.setoid _)
+  map :=  Discrete.eqToHom ∘ congrArg fct ∘ Quotient.sound ∘ Zigzag.of_hom
+
+private def fctrToFn {X} {C : Cat} (fctr :C ⥤ typeToCat.obj X)  : (connectedComponents.obj C ⟶ X) :=
```

Comment:
Similarly to above, this belongs in a different file and shouldn't mention `Cat`, `connectedComponents` or `typeToCat`. Something like

```lean
def foo {X C : Type*} [Category C] (F : C ⥤ Discrete X) : 
    (ConnectedComponents C → X) :=
  Quotient.lift
    (fun c => (F.obj c).as)
    (fun _ _ h => eq_of_zigzag X (zigzag_obj_of_zigzag F h))
```

(with a proper name, of course). Then, you can define a more general `Equiv` which will then be then applied in the `homEquiv` field of the adjunction below.